test/util/framework/zero_dep.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"hash/fnv"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	hcpapi20240610 "github.com/Azure/ARO-HCP/internal/api/v20240610preview/generated"
+)
+
+type TestContext struct {
+	subscriptionID   string
+	tenantID         string
+	testUserClientID string
+
+	contextLock               sync.RWMutex
+	azureCredentials          azcore.TokenCredential
+	clientFactory20240610     *hcpapi20240610.ClientFactory
+	armResourcesClientFactory *armresources.ClientFactory
+}
+
+type CleanupFunc func(ctx context.Context) error
+
+var (
+	invocationContext *TestContext
+	initializeOnce    sync.Once
+)
+
+// InvocationContext requires the following env vars
+// CUSTOMER_SUBSCRIPTION
+// AZURE_TENANT_ID
```

Comment:
You are best off using DefaultCredential() from the azure lib, so you are agnostic to which method (SP, MSI, etc) is being provided to auth with